elastic/logstash
# Allow metrics update when PQ draining

Fixed: #13832



## Release notes


Fixed no metrics update issue when PQ is draining

## What does this PR do?



move stopping metrics collection after pipelines shutdown to allow metrics update during PQ draining

## Why is it important/What is the impact to the user?



PQ draining could be a long process. User is unable to get latest metrics through `/_node/stats`

## Checklist




## Author's Checklist



## How to test this PR locally


- enable `queue.type: persisted` and `queue.drain: true`
```
    input {
      generator { count => 10000 }
    }
    filter {
      sleep { 
        time => "1" 
        every => 10
      }
    }
    output {
      stdout { }
    }
```
- send `SIGTERM` to Logstash to enter draining phase
- `curl -XGET 'localhost:9600/_node/stats/'`, metrics should keep update


## Related issues


- 

## Use cases



## Screenshots



## Logs
